README: Add note about unsupported compilers

Patater opened the pull request.

Patater committed:
README: Add note about unsupported compilers

jamesbeyond commented:
@Patater , this line might need to change as well:
https://github.com/ARMmbed/mbed-os-example-atecc608a#compiling-and-running-the-example
Maybe the adding line goes here might be better?

Patater committed:
README: Clarify which toolchain options are OK

The example currently only builds with GCC_ARM, so don't list ARM and
IAR yet, when listing which toolchains you can build the example with.

Patater commented:
> @Patater , this line might need to change as well:
> https://github.com/ARMmbed/mbed-os-example-atecc608a#compiling-and-running-the-example
> Maybe the adding line goes here might be better?

Good point. I've added a commit to remove mention of those other options.

jamesbeyond reviewed (approved):
looks good
